pkg/apis/migration/v1alpha1/condition.go

```diff
@@ -40,110 +51,193 @@ func (r *Condition) Update(other Condition) {
 	r.Type = other.Type
 	r.Status = other.Status
 	r.Reason = other.Reason
+	r.Category = other.Category
 	r.Message = other.Message
 	r.LastTransitionTime = metav1.NewTime(time.Now())
 }
 
 func (r Condition) Equal(other Condition) bool {
 	return r.Type == other.Type &&
 		r.Status == other.Status &&
+		r.Category == other.Category &&
 		r.Reason == other.Reason &&
 		r.Message == other.Message
 }
 
 // Managed collection of conditions.
 // Intended to be included in resource Status.
+// List - The list of conditions.
+// staged - In `staged` mode, the search methods like
+//          HasCondition() filter out un-staged conditions.
+// -------------------
+// Example:
+//
+// thing.Status.BeginStagedConditions()
+// thing.Status.SetCondition(c)
+// thing.Status.SetCondition(c)
+// thing.Status.SetCondition(c)
+// thing.Status.EndStagedConditions()
+// thing.Status.SetReady(
+//     !thing.Status.HasBlockerCondition(),
+//     "Resource Ready.")
+//
 type Conditions struct {
-	Conditions []Condition `json:"conditions"`
+	List   []Condition `json:"conditions"`
+	staged bool
+}
+
+// Begin staged conditions.
+func (r *Conditions) BeginStagedConditions() {
```

Comment:
So this sets an entire group of conditions as `staged`. What does that mean?

Reply:
Would a better name be "staging"? If this is meant to communicate that the process of staging conditions is in progress.

Reply:
Would `staging` perhaps be a better choice for the overall bool?

Reply:
This puts the `Conditions` collection in _staging_ mode and _un-stages_ all the conditions. (Condition.staged=false).